### PR TITLE
Go Modules housekeeping and README updates.

### PR DESCRIPTION
**Pull request by alain2sf**
Signed-off-by: Alain Pulluelo <alain.pulluelo@forgerock.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,7 @@ _*.sh
 vendor
 glide.lock
 *.DS_Store
+go.sum
```
Comment (anonymouse64) [inline .gitignore:10]: @tsconn23 what is the current guidance on go.sum committing? I thought that we had started to commit the go.sum and update it when the dependencies change

**Review by anonymouse64 — COMMENTED** [review]
We should check to see if the go.sum should be committed or ignored. I think it should be committed.

File: .gitignore (hunk 1 above)
Comment (alain2sf) [reply .gitignore:10]: @anonymouse64 @tsconn23 I also believe we should commit the go.sum but it saw in edgex-go repo that it wasn't so I followed this model... Thought it was maybe an already discussed decision.

Comment (alain2sf) [reply .gitignore:10]: FYI https://github.com/golang/go/wiki/Modules#releasing-modules-all-versions

Comment (anonymouse64) [reply .gitignore:10]: ah actually it was discussed in one of the devops calls : https://github.com/edgexfoundry/edgex-go/issues/1141

Comment (alain2sf) [reply .gitignore:10]: I agree with you I would like to know the rationale behind...

Comment (alain2sf) [reply .gitignore:10]: @anonymouse64 Ian, shall we (for the time being) follow the devops group recommendation (even if I'm not convinced) and complete/close that PR. Until we get more insight or a change of practice, committing the go.sum in a next future will not require much effort anyways.

**Review by anonymouse64 — APPROVED** [review]
LGTM, would be nice to see an issue about committing go.sum at some point when the project is ready to do so

**Review by tingyuz — APPROVED** [review]
LGTM. We can always add the .sum file if needed in the future.

**Pull request merged**